tslib/src/functions-throttled/school-import/data-access/school.data.service.ts

```diff
@@ -0,0 +1,65 @@
+import * as mssql from 'mssql'
+import { ILogger } from '../../../common/logger'
+import { SchoolImportJobResult } from '../SchoolImportJobResult'
+import { ISchoolRecord } from './ISchoolRecord'
+
+export interface ISchoolDataService {
+  bulkUpload (logger: ILogger, data: any, jobResult: SchoolImportJobResult): Promise<SchoolImportJobResult>
+}
+
+export class SchoolDataService implements ISchoolDataService {
+
+  private pool: mssql.ConnectionPool
+  private jobResult: SchoolImportJobResult
+
+  constructor (pool: mssql.ConnectionPool, jobResult: SchoolImportJobResult) {
+    this.pool = pool
+    this.jobResult = jobResult
+  }
+
+  private logError (msg: string): void {
+    this.jobResult.stderr.push(`${(new Date()).toISOString()} school-import: ${msg}`)
+  }
+
+  /**
+   * Perform a bulk upload to the school table, inserting new schools
+   * @param context - function context object
+   * @param schoolData - the csv parsed to array or arrays without header row
+   * @param mapping - the mapping between our domain and the input file
+   * @return {SchoolImportJobResult}
+   */
+  async bulkUpload (logger: ILogger, schoolData: Array<ISchoolRecord>): Promise<SchoolImportJobResult> {
+    logger.verbose('SchoolDataService.bulkUpload() called')
+
+    const table = new mssql.Table('[mtc_admin].[school]')
+    table.create = false
+    table.columns.add('dfeNumber', mssql.Int, { nullable: false })
+    table.columns.add('estabCode', mssql.NVarChar(mssql.MAX), { nullable: true })
```

Comment:
Why is this col (and leaCode) nullable?

Reply:
untouched from original implementation, good spot though

Reply:
leaCode & estabCode are nullable in the database, which seems odd given that dfeNumber is not.  No schools have, nor should have a null value for either of these.  Will create a ticket for that, this will have to remain as-is for now and will be remediated in the upcoming ticket.